File: sdks/ruby/lib/novabilling/types/add_on_price_response.rb
# frozen_string_literal: true

module Novabilling
  module Types
    class AddOnPriceResponse < Internal::Types::Model
      field :id, -> { String }, optional: false, nullable: false
      field :add_on_id, -> { String }, optional: false, nullable: false, api_name: "addOnId"
      field :currency, -> { String }, optional: false, nullable: false
      field :amount, -> { String }, optional: false, nullable: false
    end
  end
end

File: sdks/csharp/src/NovaBilling/Types/AddOnPriceResponse.cs
using System.Text.Json;
using System.Text.Json.Serialization;
using NovaBilling.Core;

namespace NovaBilling;

[Serializable]
public record AddOnPriceResponse : IJsonOnDeserialized
{
    [JsonExtensionData]
    private readonly IDictionary<string, JsonElement> _extensionData =
        new Dictionary<string, JsonElement>();

    [JsonPropertyName("id")]
    public required string Id { get; set; }

    [JsonPropertyName("addOnId")]
    public required string AddOnId { get; set; }

    [JsonPropertyName("currency")]
    public required string Currency { get; set; }

    /// <summary>
    /// Decimal amount as string
    /// </summary>
    [JsonPropertyName("amount")]
    public required string Amount { get; set; }

    [JsonIgnore]
    public ReadOnlyAdditionalProperties AdditionalProperties { get; private set; } = new();

    void IJsonOnDeserialized.OnDeserialized() =>
        AdditionalProperties.CopyFromExtensionData(_extensionData);

    /// <inheritdoc />
    public override string ToString()
    {
        return JsonUtils.Serialize(this);
    }
}
